gce-containers-startup/runtime/runtime.go

```diff
@@ -165,40 +165,62 @@ func pullImage(dockerClient DockerApiClient, auth string, spec api.Container) er
 	return nil
 }
 
-func findIdForName(containers []dockertypes.Container, containerName string) (string, bool) {
-	var searchName = "/" + containerName
-	for _, container := range containers {
-		for _, name := range container.Names {
-			if name == searchName {
-				return container.ID, true
-			}
-		}
-	}
-	return "", false
-}
-
-func deleteOldContainer(dockerClient DockerApiClient, containerName string) error {
+// deleteOldContainers deletes all containers started by konlet.
+// rawName is a container name without any generate prefixes or suffixes.
+func deleteOldContainers(dockerClient DockerApiClient, rawName string) error {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
 	listOpts := dockertypes.ContainerListOptions{All: true}
-	resp, err := dockerClient.ContainerList(ctx, listOpts)
-
+	containers, err := dockerClient.ContainerList(ctx, listOpts)
 	if err != nil {
 		return err
 	}
-
-	containerID, exists := findIdForName(resp, containerName)
-	if !exists {
-		log.Printf("Container with name '%s' has not yet been run.\n", containerName)
+	idsNames := containersStartedByKonlet(containers, rawName)
+	if len(idsNames) == 0 {
+		log.Print("No containers to delete.\n")
 		return nil
 	}
+	for id, name := range idsNames {
+		log.Printf("Removing previous container '%s' (ID: %s)\n", name, id)
```

Comment:
No sure "previous container" is super clear to me - maybe "container started by previous run of the script" or sth similar?

Reply:
Done.